### PR TITLE
gnmi-1.8: add NoMetadataSingleUpdate deviation

### PR DESCRIPTION
**Pull request by mojiiba**
Currently our devices require metadata update to be accompanied by another update for an OC path (no single metadata update is allowed). This deviation allows setting metadata along with another update.

### CODE REVIEW

**Review by vcheny — APPROVED** [review]

**Review by vosipchu — APPROVED** [review]

File: internal/deviations/deviations.go [hunk 1]
```diff
@@ -112,5 +112,7 @@ var (
 	ExplicitInterfaceInDefaultVRF = flag.Bool("deviation_explicit_interface_in_default_vrf", false,
 		"Device requires explicit attachment of an interface or subinterface to the default network instance. OpenConfig expects an unattached interface or subinterface to be implicitly part of the default network instance. Fully-compliant devices should pass with and without this deviation.")
 
-	ExplicitPortSpeed = flag.Bool("deviation_explicit_port_speed", false, "Device requires port-speed to be set because its default value may not be usable. Fully compliant devices should select the highest speed available based on negotiation.")
+	NoMetadataSingleUpdate = flag.Bool("deviation_nometadata_single_update", false, "Device requires metadata update to be accompanied by another update for an OC path.")
```
Comment (thesrinath) [inline internal/deviations/deviations.go:115]: As discussed, let's make this the default behaviour so this file can be reverted.

File: feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go [hunk 2]
```diff
@@ -100,6 +104,14 @@ func TestGNMIMetadataAnnotation(t *testing.T) {
 			t.Errorf("Cannot build a gNMI SetRequest from proto message: %v", err)
 		}
 
+		if *deviations.NoMetadataSingleUpdate {
```
Comment (thesrinath) [inline feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go:107]: remove if and make the standard behaviour

**Comment by alexmasi** [comment]
good for go

File: feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go [hunk 3]
```diff
@@ -217,3 +229,24 @@ func extractMetadataAnnotation(getResponse *gpb.GetResponse, m proto.Message) er
 	}
 	return nil
 }
+
+// buildGNMIUpdate builds a gnmi update for a given ygot path and value.
+func buildGNMIUpdate(t *testing.T, yPath ygnmi.PathStruct, val interface{}) *gpb.Update {
```
[the remark this answers is not in the document]
Comment (mojiiba) [reply feature/system/gnmi/metadata/tests/annotation_test/annotation_test.go:235]: done, i also changed the other helpers and remove replace the fmt.print with t.log

**Review by thesrinath — APPROVED** [review]

**Pull request merged**